ls1intum/Artemis
# `Tutorial groups`: Fix schedule and sessions recreated when any tutorial group property changed

### Checklist
#### General
#### Server

### Motivation and Context
There was a bug in the server logic that caused the schedule and sessions to be recreated every time any tutorial group property was updated even if its totally unrelated to the schedule or sessions.

### Description
Fixed code, such that tutorial group property changes unrelated to the schedule and exclusive schedule location changes do not cause a recreation of sessions

### Steps for Testing
- Test that updating a tutorial group property unrelated to the schedule (such as title) should not cause schedule and sessions to be recreated. You can check this by noting down the session ids in the session management table, and check that they are not changed after the update. Also any modification of you to the sessions should not be affected by the update.
- Test that updating just the location of a schedule does not recreate the schedule or the sessions. Check again with the ids like above. 


### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
- Added test that covers the bugs.


| Component                    | Statement Coverage | Branch Coverage |
|------------------------------|--------------------|-----------------|
| TutorialGroupScheduleService | 85%                | 80%             |
